app/types/type-parser.pegjs

```diff
@@ -0,0 +1,66 @@
+// This represents a scope
+// currently it doesn't make any sense to have a nested scope
+
+Definitions = _ "{" _ dfns:Definition* _ "}" _ { return new types.TypeScope(new Map(dfns)); }
```

Comment:
If we are making our own language,

```
return new types.TypeScope(new Map(dfns));
```
Seems much too verbose. Maintaining similarity to JavaScript is good, but we should take advantage of having our own parser where possible. For instance, at the very least we should remove the both "new"s from the spec.